src/blockdiagonal.jl

```diff
@@ -102,6 +107,23 @@ end
 
 ## Base
 Base.Matrix(B::BlockDiagonal) = cat(blocks(B)...; dims=(1, 2))
+
+function ChainRulesCore.rrule(::Type{<:Base.Matrix}, B::BlockDiagonal)
+    function Matrix_pullback(Δ::AbstractMatrix)
+        nrows = size.(B.blocks, 1)
+        ncols = size.(B.blocks, 2)
+        row_idxs = cumsum(nrows) .- nrows .+ 1
+        col_idxs = cumsum(ncols) .- ncols .+ 1
+
+        Δblocks = [
+            Δ[row_idxs[n]:(row_idxs[n] + nrows[n] - 1), col_idxs[n]:(col_idxs[n] + ncols[n] - 1)] 
+            for n in eachindex(B.blocks)
+        ]
```

Comment:
I think it might be clearer to write:
```suggestion
        Δblocks = map(eachindex(B.blocks)) do n
            block_rows = row_idxs[n]:(row_idxs[n] + nrows[n] - 1)
            block_cols = col_idxs[n]:(col_idxs[n] + ncols[n] - 1)
            return Δ[block_rows, block_cols] 
        end
```

or even:
```suggestion
        Δblocks = map(eachindex(B.blocks)) do n
            block_rows = row_idxs[n] .+ (0:nrows[n] - 1)
            block_cols = col_idxs[n] .+ (0:ncols[n] - 1)
            return Δ[block_rows, block_cols] 
        end
```